csrc/fusion.cpp

```diff
@@ -205,8 +205,20 @@ void Fusion::removeVal(Val* val) {
     removeExpr(orig);
   }
 
-  for (Expr* use : unordered_uses(val)) {
-    removeExpr(use);
+  const std::unordered_set<Expr*> exprs = exprs_;
```

Comment:
I don't think `exprs` is used anywhere. 

```suggestion
```

Reply:
Ah thanks for catching this. I was previously copying the whole set to prevent altering it inside the loop.